homeassistant/components/light/knx.py

```diff
@@ -154,19 +163,45 @@ def supported_features(self):
         if self.device.supports_brightness:
             flags |= SUPPORT_BRIGHTNESS
         if self.device.supports_color:
-            flags |= SUPPORT_COLOR
+            flags |= SUPPORT_COLOR | SUPPORT_BRIGHTNESS
         return flags
 
     async def async_turn_on(self, **kwargs):
         """Turn the light on."""
+        # initialize with current values
+        brightness = self.brightness
+        hs_color = self.hs_color
+
+        # overwrite values with new settings, if available
         if ATTR_BRIGHTNESS in kwargs:
-            if self.device.supports_brightness:
-                await self.device.set_brightness(int(kwargs[ATTR_BRIGHTNESS]))
-        elif ATTR_HS_COLOR in kwargs:
-            if self.device.supports_color:
-                await self.device.set_color(color_util.color_hs_to_RGB(
-                    *kwargs[ATTR_HS_COLOR]))
+            brightness = int(kwargs[ATTR_BRIGHTNESS])
+        if ATTR_HS_COLOR in kwargs:
+            hs_color = kwargs[ATTR_HS_COLOR]
```

Comment:
There is some lint left: `homeassistant/components/light/knx.py:178:8: R1715: Consider using dict.get for getting values from a dict if a key is present or a default if not (consider-using-get)`

So the proposal is to use `hs_color = kwargs.get(ATTR_HS_COLOR, self.hs_color)`